aggregator/placement_mgr.go

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package aggregator
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/m3db/m3cluster/placement"
+	"github.com/m3db/m3cluster/shard"
+	"github.com/m3db/m3x/clock"
+	"github.com/uber-go/tally"
+)
+
+var (
+	errPlacementManagerNotOpenOrClosed = errors.New("placement manager not open or closed")
+	errPlacementManagerOpenOrClosed    = errors.New("placement manager already open or closed")
+	errInstanceNotFoundInPlacement     = errors.New("instance not found in placement")
```

Comment:
Should this error be public since it's being checked by the user of PlacementManager

Reply:
Sure thing.